### PR TITLE
pipeline: change event processing to be map based

### PR DESCRIPTION
**Pull request by NDStrahilevitz**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

```
Author: Nadav Strahilevitz <nadav.strahilevitz@aquasec.com>
Date:   Wed Dec 21 14:00:32 2022 +0000

    pipeline: change event processing to be map based
    
    The event processing part of the pipeline's mechanism is changed from a
    hard coded switch statement to use a map which registers for each event
    id it's processing functions.
    By using this mechanism we can expose a function to register such a
    processor programatically, which will be useful later for future tracee
    event plugins.
```

Fixes: #issue_number

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [x] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Did a manual sanity check with `hooked_proc_fops` and the `phide` rootkit which is an event going through processing.

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
I want to review this one before it gets merged.

**Review by yanivagman — COMMENTED** [review]
Nice change!

File: pkg/ebpf/events_pipeline.go
Comment (NDStrahilevitz) [reply pkg/ebpf/events_pipeline.go:266]: Right, i'll change it. Should be good to merge afterwards right?

Comment (yanivagman) [reply pkg/ebpf/events_pipeline.go:266]: Yes. Needs rebase as well.

**Review by yanivagman — APPROVED** [review]
LGTM

**Comment by NDStrahilevitz** [comment]
Waiting for a review from @rafaeldtinoco before merging since he requested it.

**Comment by rafaeldtinoco** [comment]
Spoke to @NDStrahilevitz and we agreed to split the processing functions in different files (like derivation code). That will make things look cleaner for big processing functions (like vfs related ones).

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
Per last comment.

**Pull request merged**
